main/src/ca/uwaterloo/flix/language/phase/jvm/GenFunAndClosureClasses.scala

```diff
@@ -117,6 +120,23 @@ object GenFunAndClosureClasses {
     *   }
     * }
     * }}}
+    *
+    * For control-pure functions, the generated code deviates slightly, as the `applyFrame` method is replaced
+    * by a static method called `directApply` that accepts the concrete parameters of the Flix function.
+    * The `invoke` method then forwards its call to `directApply`, passing the values from the corresponding fields.
+    *
+    * {{{
+    * public final class Def$example extends Fn2$Obj$Int$Obj implements Frame {
+    *
+    *   public final Result invoke() { return this.applyFrame(null); }
```

Comment:
So the body here is really `return applyDirect(this.arg0, this.arg1)`?

Reply:
Oops yes